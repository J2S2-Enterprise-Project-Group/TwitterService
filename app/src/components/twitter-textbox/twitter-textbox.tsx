import { TextareaAutosize } from "@material-ui/core";
import React from "react";

interface TwitterTextboxProps {
    onBlur: (evt: React.ChangeEvent<any>) => void
}
export const TwitterTextbox: React.FC<TwitterTextboxProps> = ({ onBlur }) => {
    return (
        <React.Fragment>
            <TextareaAutosize
                aria-label="Whats Happening"
                rowsMin="5"
                maxLength={140}
                placeholder="Whats Happening"
                style={{ 
                    width: '95%',
                    padding:'10px',
                    fontSize:'14px',
                    fontFamily:'Roboto'
                 }}
                onBlur={onBlur} 
                />
        </React.Fragment>
    );
}