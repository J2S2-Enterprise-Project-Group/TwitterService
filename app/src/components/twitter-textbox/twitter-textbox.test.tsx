/**
 * @author : Shiv Kumar Ganesh<shivkumar.ganesh@sjsu.edu>
 */
import React from 'react';
import { ReactWrapper, shallow } from 'enzyme';
import { TwitterTextbox } from './twitter-textbox';
import { act } from 'react-dom/test-utils';
import { TextareaAutosize } from '@material-ui/core';

const title = 'Twitter App';
const onBlur = jest.fn();
let wrapped = shallow(<TwitterTextbox onBlur={onBlur} />);
describe('Twitter TextBox', () => {
    it('Check if the component loaded', () => {
        expect(wrapped).toMatchSnapshot();
    });

    it('renders the TextBox', () => {
        expect(wrapped.find(TextareaAutosize)).toHaveLength(1);
    });

    it('twigger method onBlur when blurred', () => {
        const blurEvent = { event: { target: 'Hello World Tweet' } }
        act(() => {
            wrapped.find(TextareaAutosize).simulate('blur', blurEvent);
        });
        expect(onBlur).toBeCalledTimes(1)
        expect(onBlur).toBeCalledWith(blurEvent)
    })
});