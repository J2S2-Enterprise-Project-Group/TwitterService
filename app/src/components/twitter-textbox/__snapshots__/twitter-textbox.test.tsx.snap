// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Twitter TextBox Check if the component loaded 1`] = `
<Fragment>
  <ForwardRef(TextareaAutosize)
    aria-label="Whats Happening"
    maxLength={140}
    onBlur={[MockFunction]}
    placeholder="Whats Happening"
    rowsMin="5"
    style={
      Object {
        "fontFamily": "Roboto",
        "fontSize": "14px",
        "padding": "10px",
        "width": "95%",
      }
    }
  />
</Fragment>
`;
